### ISSUE TITLE
App Bar disapears in "Share Dialog"

### ISSUE THREAD

**Issue by Sputnick85** [nextcloud/android#2883]
### Actual behaviour
The App Bar in this dialogue disapers and apperas while making changes in the Window.

- Selecting a File to share
![image001](https://user-images.githubusercontent.com/3588747/44033381-7ed77960-9f0a-11e8-951d-785120bc7aa6.jpg)

- Choosing the Link Symbole
![image002](https://user-images.githubusercontent.com/3588747/44033382-7ef72ddc-9f0a-11e8-81fd-3279d6d9c660.jpg)

- _the App Bar is not realy present(red box)_
- _the back arrow is pesent but in this specific case hard to see(yellow circle)_
- choosing the "ShareLink" option
![image005](https://user-images.githubusercontent.com/3588747/44033383-7f2a3196-9f0a-11e8-8e38-291171178cd7.jpg)

- _in the background the App Bar is present again(red circle)_
- Choosing "CopyLink"
![image006](https://user-images.githubusercontent.com/3588747/44033384-7f45de6e-9f0a-11e8-881c-2eabe5eff65b.jpg)

- _the App Bar disapers again, back arrow is still present(red circle)_
- deactivating the share permission again
![image007](https://user-images.githubusercontent.com/3588747/44033385-7f6c580a-9f0a-11e8-8f77-adc221c1e3a9.jpg)

- _the AppBar is present again(red circle)_
![image008](https://user-images.githubusercontent.com/3588747/44033380-7eb0f95c-9f0a-11e8-8bfe-2d69902029fa.jpg)

### Expected behaviour
The App Bar should be presant all the time.
 
### Steps to reproduce
1. Select a file to share via Link
2. now the AppBar is gone
3. the App Bar will apear again if you toggle the "share checkbox"
4. if you turn the Display off and on again the AppBar will disapear again

### Environment data
Android version:
8.0.0
Device model: 
Samsung Galaxy S8
Stock or customized system:
Stock
Nextcloud app version:
3.2.1
Nextcloud server version:
13.0.5

**Comment by AndyScherzinger** [comment]
There is indeed an UX here which is rather about the bar icons not being very easy to seee depending on the image.

> The App Bar should be presant all the time.

The app bar is transparent for images which is on purpose so the issue is rather about not being able to easily spot the controls within the bar. We'll have to think about how this can be improved.

Any ideas/comments @tobiasKaminsky @nextcloud/designers ?

**Comment by tobiasKaminsky** [comment]
The latest image shows the app bar although it is an image, so this should be easy to fix, I hope.
Regarding the other one, I guess we need to use something like this https://developer.android.com/training/material/palette-colors:

- limit the bitmap to the area where the "<-" back arrow is located
- get e.g. dominantSwatch (https://developer.android.com/reference/android/support/v7/graphics/Palette.html#getDominantSwatch())
- toggle white/black/even another color depending on color above

**Comment by AndyScherzinger** [comment]
Using palettes is one thing I also thought about while this will generate the values based on the whole image so it'll be quite okay but will still leave us with edge cases (which I think we need to live with). It works quite fine if:

> limit the bitmap to the area where the "<-" back arrow is located

but that might be really tricky to do.

**Comment by tobiasKaminsky** [comment]
It does not need to be pixel perfect, I think.

Here we should get a rough size, and knowing that it start from top-left corner, we can extract an area.
https://github.com/nextcloud/android/blob/4f7fcf2bb58f0b53f0e2ad9ce5fc807092dc1166/src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java#L89-L91

(in example above, it would be enough to know that this area is a dark green (tree) and therefore tint the arrow white)

**Comment by AndyScherzinger** [comment]
> (in example above, it would be enough to know that this area is a dark green (tree) and therefore tint the arrow white)

Sounds good to me :+1:

**Comment by jancborchardt** [comment]
Oh cool, so making the arrow black or white depending on the image background would be possible?

Another possibility is also to do it similar to Netflix, what seems to be:
- Having the icons in white always, and a very slight dark shadow maybe
- A short fade of slightly transparent dark on the top to transparency further down
![](https://lh3.googleusercontent.com/jqdEoSdMRyEcuqOpE7DACHYWJAB-A1RgqnNSQGarjJ6rAM2c0aUlgGV4peEYneQ6en8=w2240-h1124)
Basically any photo viewer app also does this. And that also usually works with every kind of image.

What’s important for when the bar is visible (and intented to be visible) is that all the icons and text have the same color. Either all black or all white. There’s some screenshots where everything is black but the back arrow is white.

**Comment by jancborchardt** [comment]
Issue is still current, as is my feedback above. :) Several options (which are mutually exclusive):
- The white back arrow gets a slight drop shadow
- The screen gets a short fade from dark to transparent on the top, as shown in the Netflix screenshot above
- The back arrow gets a circle around it for better visibility (least good looking option)

**Comment by tobiasKaminsky** [comment]
> * The white back arrow gets a slight drop shadow

If I provide you with the svg image for this arrow, could you add this?

**Comment by jancborchardt** [comment]
@tobiasKaminsky sure. :) Meanwhile on the server we do this via CSS – it’s not possible to do it programmatically on Android?

**Comment by tobiasKaminsky** [comment]
> > *
> 
> If I provide you with the svg image for this arrow, could you add this?

@jancborchardt can you do this for us? :heart:

**Comment by AndyScherzinger** [comment]
@jancborchardt @tobiasKaminsky I created a PR adding a gradient, please review, _if_ fine by you (gradient darkness can be discussed of course) than we don't need a drop-shadowed-back-arrow :)

For screenshots see #5634
